public/app/features/explore/utils/links.test.ts

```diff
@@ -16,470 +16,574 @@ import { initTemplateSrv } from '../../../../test/helpers/initTemplateSrv';
+      const { field, range } = setup({
+        title: '',
+        url: '',
+        internal: {
+          query: { query: 'query_1' },
+          datasourceUid: 'uid_1',
+          datasourceName: 'test_ds',
+          panelsState: {
+            trace: {
+              spanId: 'abcdef',
+            },
+          },
+        },
+      });
+      const splitfn = jest.fn();
+      const links = getFieldLinksForExplore({
+        field,
+        rowIndex: ROW_WITH_TEXT_VALUE.index,
+        splitOpenFn: splitfn,
+        range,
+      });
+
+      expect(links[0].href).toBe(
+        `/explore?left=${encodeURIComponent(
+          '{"range":{"from":"now-1h","to":"now"},"datasource":"uid_1","queries":[{"query":"query_1"}],"panelsState":{"trace":{"spanId":"abcdef"}}}'
+        )}`
+      );
+      expect(links[0].title).toBe('test_ds');
+
+      if (links[0].onClick) {
+        links[0].onClick({});
+      }
+
+      expect(splitfn).toBeCalledWith({
         datasourceUid: 'uid_1',
-        datasourceName: 'test_ds',
+        query: { query: 'query_1' },
+        range,
         panelsState: {
           trace: {
             spanId: 'abcdef',
           },
         },
-      },
+      });
     });
-    const splitfn = jest.fn();
-    const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_TEXT_VALUE.index, splitOpenFn: splitfn, range });
-
-    expect(links[0].href).toBe(
-      `/explore?left=${encodeURIComponent(
-        '{"range":{"from":"now-1h","to":"now"},"datasource":"uid_1","queries":[{"query":"query_1"}],"panelsState":{"trace":{"spanId":"abcdef"}}}'
-      )}`
-    );
-    expect(links[0].title).toBe('test_ds');
-
-    if (links[0].onClick) {
-      links[0].onClick({});
-    }
-
-    expect(splitfn).toBeCalledWith({
-      datasourceUid: 'uid_1',
-      query: { query: 'query_1' },
-      range,
-      panelsState: {
-        trace: {
-          spanId: 'abcdef',
+
+    it('returns correct link model for external link when user does not have access to explore', () => {
+      const { field, range } = setup(
+        {
+          title: 'external',
+          url: 'http://regionalhost',
         },
-      },
-    });
-  });
+        false
+      );
+      const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_TEXT_VALUE.index, range });
 
-  it('returns correct link model for external link when user does not have access to explore', () => {
-    const { field, range } = setup(
-      {
-        title: 'external',
-        url: 'http://regionalhost',
-      },
-      false
-    );
-    const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_TEXT_VALUE.index, range });
+      expect(links[0].href).toBe('http://regionalhost');
+      expect(links[0].title).toBe('external');
+    });
 
-    expect(links[0].href).toBe('http://regionalhost');
-    expect(links[0].title).toBe('external');
-  });
+    it('returns no internal links if when user does not have access to explore', () => {
```

Comment:
This will fail now because `replace` uses FieldProxy that returns an empty string for non-existing fields. I think it's fine because we want to encourage users to use shorter access variables. I think we can just have two tests showing that behavior:

```typescript
    it('returns internal links for non-existing fields accessed with __data.fields', () => {
      const { field, range, dataFrame } = setup({
        title: '',
        url: '',
        internal: {
          query: { query: 'query_1-${__data.fields.flux-dimensions}' },
          datasourceUid: 'uid_1',
          datasourceName: 'test_ds',
        },
      });
      const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_NULL_VALUE.index, range, dataFrame });
      expect(links).toHaveLength(1);
    });

    it('returns no internal links when target contains empty template variables', () => {
      const { field, range, dataFrame } = setup({
        title: '',
        url: '',
        internal: {
          query: { query: 'query_1-${mementoMori}' },
          datasourceUid: 'uid_1',
          datasourceName: 'test_ds',
        },
      });
      const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_NULL_VALUE.index, range, dataFrame });
      expect(links).toHaveLength(0);
    });
```